src/pools/selection-strategies.ts

```diff
@@ -0,0 +1,118 @@
+import type { JSONValue, MessageValue } from '../utility-types'
+import { isKillBehavior, KillBehaviors } from '../worker/worker-options'
+import type { AbstractPool, IWorker } from './abstract-pool'
+
+/**
+ * Result of the round robin selection function.
+ *
+ * @template Type of the chosen element.
+ */
+export interface RoundRobinSelectionResult<Element> {
```

Comment:
You want to define a type for each selection algo?

Reply:
No, this was just for easier JSDoc and typedef
It is also not exported via `index.ts` by design